### PR TITLE
Add plain page

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Add plain page and binary plain page for string

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/plain_page.h [hunk 1]
```diff
@@ -0,0 +1,170 @@
+#pragma once
```
Comment (imay) [inline be/src/olap/rowset/segment_v2/plain_page.h]: License header

File: be/src/olap/rowset/segment_v2/plain_page.h [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    //     release() should be called after finish
+    //     reset() should be called after this function before reuse the builder
+    void release() override {
+        uint8_t *ret = _buffer.release();
+        (void) ret;
+    }
+
+private:
+    faststring _buffer;
+    const PageBuilderOptions *_options;
```
Comment (imay) [inline be/src/olap/rowset/segment_v2/plain_page.h]: you should copy this option. This option maybe a variable in stack, which would be invalid when you use later

File: be/src/olap/rowset/segment_v2/plain_page.h [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        return Status::OK();
+    }
+
+    Status get_dictionary_page(Slice *dictionary_page) override {
+        return Status::NotSupported("get_dictionary_page not supported in plain page builder");
+    }
+
+    Slice finish(const rowid_t page_first_rowid) override {
+        encode_fixed32_le((uint8_t *) &_buffer[0], _count);
+        encode_fixed32_le((uint8_t *) &_buffer[4], page_first_rowid);
```
Comment (imay) [inline be/src/olap/rowset/segment_v2/plain_page.h]: first rowid is parsed in ColumnReader. And this page only store data, if there is null this first row id is strange here

File: be/src/olap/rowset/segment_v2/plain_page.h [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        return Slice(_buffer.data(),  plainPageHeaderSize + _count * SIZE_OF_TYPE);
+    }
+
+    void reset() override {
+        _count = 0;
+        _buffer.clear();
+        _buffer.resize(plainPageHeaderSize);
```
Comment (imay) [inline be/src/olap/rowset/segment_v2/plain_page.h]: you should reserve enough space for data to avoid needless data copy

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/segment_v2/plain_page.h [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    Status next_batch(size_t *n, ColumnVectorView *dst) override {
+        DCHECK(_parsed);
+
+        if (PREDICT_FALSE(*n == 0 || _cur_idx >= _num_elems)) {
+            *n = 0;
+            return Status::OK();
+        }
+
+        size_t max_fetch = std::min(*n, static_cast<size_t>(_num_elems - _cur_idx));
+        memcpy(dst->column_vector()->col_data(),
```
Comment (kangpinghuang) [inline be/src/olap/rowset/segment_v2/plain_page.h]: should consider row_offset in ColumnVectorView here.
And How to process string and binary values?
It seems that you should store value length for string or binary data?

File: be/test/olap/rowset/segment_v2/plain_page_test.cpp [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+}
+}
+
+int main(int argc, char** argv) {
+    ::testing::InitGoogleTest(&argc, argv);
+    return RUN_ALL_TESTS();
+}
+
+
```
Comment (kangpinghuang) [inline be/test/olap/rowset/segment_v2/plain_page_test.cpp]: remove this additional empty line

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

**Review by imay — CHANGES_REQUESTED** [review]
you should add unit test in run-ut.sh files

**Review by imay — DISMISSED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**